File: 8.Best-Practice-e-trucchi-avanzati/objc/CM_Picker/CM_Picker WatchKit Extension/StackController.m
//
//  StackController.m
//  CM_Picker
//
//  Created by WhiteTiger "sgama la rete" on 28/02/16.
//  Copyright © 2016 WhiteTiger "sgama la rete". All rights reserved.
//

#import "StackController.h"

@interface StackController ()

@property (nonatomic, weak) IBOutlet WKInterfacePicker *picker;

@end

@implementation StackController

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];
    
    [self createPickerItems];
}

- (void)willActivate {
    [super willActivate];
}

- (void)didDeactivate {
    [super didDeactivate];
}

- (void)createPickerItems {
    NSMutableArray<WKPickerItem *> *pickerItems = [NSMutableArray array];
    
    for (int i=1; i<5; i++) {
        WKPickerItem *pickerItem = [WKPickerItem new];
        
        [pickerItem setContentImage:[WKImage imageWithImageName:[NSString stringWithFormat:@"card_0%d", i]]];
        [pickerItems addObject:pickerItem];
    }
    
    [self.picker setItems:pickerItems];
}

@end

File: 8.Best-Practice-e-trucchi-avanzati/swift/CM_Picker/CM_Picker WatchKit Extension/StackController.swift
//
//  Copyright © 2015 WhiteTiger "sgama la rete". All rights reserved.
//

import WatchKit
import Foundation

class StackController: WKInterfaceController {

    @IBOutlet var picker: WKInterfacePicker!
    
    override func awakeWithContext(context: AnyObject?) {
        super.awakeWithContext(context)
        
        self.createPickerItems()
    }

    override func willActivate() {
        super.willActivate()
    }

    override func didDeactivate() {
        super.didDeactivate()
    }

    // MARK: Private Methods
    
    private func createPickerItems() {
        var pickerItems = [WKPickerItem]()
        
        for index in 1...4 {
            let pickerItem = WKPickerItem()
            
            pickerItem.contentImage = WKImage(imageName: "card_0\(index).png")
            pickerItems.append(pickerItem)
        }
        
        picker.setItems(pickerItems)
    }
    
    @IBAction func actionPicker(value: Int) {
        
    }
    
}
